rust/lancedb/examples/bedrock.rs

```diff
@@ -0,0 +1,96 @@
+// --8<-- [start:imports]
+
+use std::{iter::once, sync::Arc};
+
+use arrow_array::{Float64Array, Int32Array, RecordBatch, RecordBatchIterator, StringArray};
+use arrow_schema::{DataType, Field, Schema};
+use futures::StreamExt;
+use aws_config::Region;
+use aws_sdk_bedrockruntime::Client;
+use lancedb::{
+    arrow::IntoArrow,
+    connect,
+    embeddings::{bedrock::BedrockEmbeddingFunction, EmbeddingDefinition,EmbeddingFunction},
+    query::{ExecutableQuery, QueryBase},
+    Result,
+};
+
+// --8<-- [end:imports]
+
+// --8<-- [start:bedrock_embeddings]
+#[tokio::main]
+async fn main() -> Result<()> {
+    let tempdir = tempfile::tempdir().unwrap();
+    let tempdir = tempdir.path().to_str().unwrap();
+    
+    // 创建 Bedrock 嵌入函数
+    let region: String ="us-east-1".to_string();
+        let config = aws_config::defaults(aws_config::BehaviorVersion::latest())
+            .region(Region::new(region))
+            .load()
+            .await;
+    
+    let embedding = Arc::new(BedrockEmbeddingFunction::new(
+        Client::new(&config)          // AWS 区域
+        //"amazon.titan-embed-text-v1", // Bedrock 模型 ID
+    ));
+
+    let db = connect(tempdir).execute().await?;
+    db.embedding_registry()
+        .register("bedrock", embedding.clone())?;
+
+    let table = db
+        .create_table("vectors", make_data())
+        .add_embedding(EmbeddingDefinition::new(
+            "text",
+            "bedrock",
+            Some("embeddings"),
+        ))?
+        .execute()
+        .await?;
+
+    // 执行向量搜索
+    let query = Arc::new(StringArray::from_iter_values(once("something warm")));
+    let query_vector = embedding.compute_query_embeddings(query)?;
+    let mut results = table
+        .vector_search(query_vector)?
+        .limit(1)
+        .execute()
+        .await?;
+
+    let rb = results.next().await.unwrap()?;
+    let out = rb
+        .column_by_name("text")
+        .unwrap()
+        .as_any()
+        .downcast_ref::<StringArray>()
+        .unwrap();
+    let text = out.iter().next().unwrap().unwrap();
+    println!("Closest match: {}", text);
+    Ok(())
+}
+// --8<-- [end:bedrock_embeddings]
```

Comment:
No need for these comments, it's used for docs
see `docs/src/basic.md` for example

Reply:
already remove these comments ,  thanks.